publication-doi-commons/src/test/java/no/unit/nva/publication/doi/PublicationMapperTest.java

```diff
@@ -0,0 +1,114 @@
+import no.unit.nva.publication.doi.dto.PublicationDate;
+import no.unit.nva.publication.doi.dto.PublicationStreamRecordTestDataGenerator;
+import no.unit.nva.publication.doi.dto.PublicationType;
+import no.unit.nva.publication.doi.dynamodb.dao.Identity;
+import nva.commons.utils.IoUtils;
+import nva.commons.utils.JsonUtils;
+import org.junit.jupiter.api.Test;
+
+class PublicationMapperTest {
+    public static final String EXAMPLE_PREFIX = "http://example.net/nva/publication/";
+    public static final String UNKNOWN_DYNAMODB_STREAMRECORD_TYPE = "UnknownType";
+
+    private static final Faker FAKER = new Faker();
+    private static final ObjectMapper objectMapper = JsonUtils.objectMapper;
+    private static final Faker faker = new Faker();
+    private static final PublicationDate EXAMPLE_PUBLICATION_DATE = new PublicationDate("1999", null, null);
+    private static final String EXAMPLE_IDENTIFIER = "654321";
+    private static final URI EXAMPLE_ID = URI.create(EXAMPLE_PREFIX + EXAMPLE_IDENTIFIER);
+    private static final PublicationType EXAMPLE_PUBLICATION_TYPE = PublicationType.JOURNAL_ARTICLE;
+    private static final String EXAMPLE_PUBLICATION_MAIN_TITLE = "Conformality loss and quantum criticality in "
+        + "topological Higgs electrodynamics in 2+1 dimensions";
+    private static final URI EXAMPLE_DOI = URI.create("https://doi.org/10.1103/physrevd.100.085005");
+    private static final URI EXAMPLE_INSTITUTION_OWNER = URI.create(
+        "https://api.dev.nva.aws.unit.no/customer/f54c8aa9-073a-46a1-8f7c-dde66c853934");
+    private static final String EXAMPLE_CONTRIBUTOR_NAME = "Nogueira, Flavio S.";
+    private static final String EXAMPLE_CONTRIBUTOR_ARPID = "451001";
+
+    private String getExampleResource(String resource) {
+        return IoUtils.stringFromResources(Path.of(resource));
+    }
+
+    private PublicationStreamRecordTestDataGenerator generateDynamoDbWithoutContributorIdentityNames() {
+        return PublicationStreamRecordTestDataGenerator.Builder.createValidPublication(FAKER)
+            .withContributorIdentities(getContributorIdentities(true))
+            .build();
+    }
+
+    private List<Identity> getContributorIdentities(boolean withoutName) {
+        List<Identity> contributors = new ArrayList<>();
+        for (int i = 0; i < faker.random().nextInt(1, 10); i++) {
+            Identity.Builder builder = new Identity.Builder();
+            builder.withArpId(faker.number().digits(10));
+            builder.withOrcId(faker.number().digits(10));
+            builder.withName(withoutName ? null : faker.superhero().name());
+            contributors.add(builder.build());
+        }
+        return contributors;
+    }
+
+    private ArrayList<URI> getIdOptions() {
+        ArrayList<URI> idOptions = new ArrayList<>();
+        idOptions.add(URI.create("https://example.net/foo/" + UUID.randomUUID().toString()));
+        idOptions.add(null);
+        return idOptions;
+    }
+
+    @Test
```

Comment:
Can you move all the tests to the top and helper methods to the bottom?

Reply:
Fixed.